aiohttp/web_fileresponse.py

```diff
@@ -122,29 +132,31 @@ async def _precondition_failed(
         return await super().prepare(request)
 
     def _get_file_path_stat_and_gzip(
-        self, check_for_gzipped_file: bool
-    ) -> Tuple[pathlib.Path, os.stat_result, bool]:
-        """Return the file path, stat result, and gzip status.
+        self, accept_encoding: str
+    ) -> Tuple[pathlib.Path, os.stat_result, Optional[str]]:
+        """Return the file path, stat result, and possible compression type.
 
         This method should be called from a thread executor
         since it calls os.stat which may block.
         """
         filepath = self._path
-        if check_for_gzipped_file:
-            gzip_path = filepath.with_name(filepath.name + ".gz")
-            try:
-                return gzip_path, gzip_path.stat(), True
-            except OSError:
-                # Fall through and try the non-gzipped file
-                pass
-
-        return filepath, filepath.stat(), False
+        for extension, encoding in ENCODING_EXTENSIONS.items():
+            if encoding in accept_encoding:
+                compressed_path = filepath.with_name(filepath.name + extension)
+                try:
+                    return compressed_path, compressed_path.stat(), encoding
+                except OSError:
+                    # Try the next extension
+                    pass
```

Comment:
Could you use `contextlib.suppress()` for clarity, instead?
```suggestion
                with suppress(OSError):
                    return compressed_path, compressed_path.stat(), encoding
```